pkg/operator/resourcesynccontroller/interfaces.go

```diff
@@ -4,6 +4,7 @@ package resourcesynccontroller
 type ResourceLocation struct {
```

Comment:
This struct is also used in the ResourceSyncController. You need to honor the value in that controller to.

Reply:
here https://github.com/openshift/library-go/blob/4a44f9500348c607b3ab376637c3126a6081715f/pkg/operator/resourcesynccontroller/resourcesync_controller.go#L155-L156

Reply:
When source is empty, this reads the cache to see if destination is present and and if it is present, it issues a delete call. In what way would you expect `Required` concept to change that flow?